app/services/task_service/gpt_generate_tests.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module TaskService
+  class GptGenerateTests < ServiceBase
+    def initialize(task:)
+      super()
+      @task = task
+      @message = task.description
+      @language = @task.programming_language.language || 'Python'
+    end
+
+    def execute
+      file_content = gpt_response
+      test_file = TaskFile.new(content: file_content, name: 'test.py', used_by_grader: true, visible: false, xml_id: SecureRandom.uuid)
+      testing_framework = TestingFramework.find_or_initialize_by(name: 'Pytest', version: 6)
+      test = Test.new(task: @task, title: 'AI-generated Test', xml_id: SecureRandom.uuid, files: [test_file], testing_framework:)
```

Comment:
This ('AI-generated Test') seems to be a translatable string, should we localise it?

Reply:
I created a localized message for it.

en
```
tests:
    generated_test: AI-generated Unit Test
```

de
```
tests:
   generated_test: 'AI-generierter Unit-Test'
```

Reply:
Let's use "KI-generierter Unit-Test" in German :+1: